src/bin/coldsnap/main.rs

```diff
@@ -35,9 +32,59 @@ async fn main() {
 
 async fn run() -> Result<()> {
     let args: Args = argh::from_env();
+    let config: aws_config::ConfigLoader = match (args.region, &args.profile) {
+        (Some(region), Some(_profile)) => {
+            //region option passed in
+            aws_config::from_env().region(Region::new(region))
+        }
+        (Some(region), None) => {
+            //region option passed in
+            aws_config::from_env().region(Region::new(region))
+        }
+        (None, Some(profile)) => {
+            //Take region from profile
+            aws_config::from_env().region(
+                aws_config::profile::ProfileFileRegionProvider::builder()
+                    .profile_name(profile)
+                    .build(),
+            )
+        }
+        (None, None) => {
+            //No region or profile passed in, use defaults
+            aws_config::from_env()
+        }
+    };
+
+    let config: aws_config::ConfigLoader = match &args.profile {
+        Some(profile) => {
+            // add profile credential provider
+            config.credentials_provider(
+                aws_config::profile::ProfileFileCredentialsProvider::builder()
+                    .profile_name(profile)
+                    .build(),
+            )
+        }
+        None => {
+            // keep config equal to itself
+            config
+        }
+    };
```

Comment:
Slightly golfed. Are the type annotations necessary?

```suggestion
    let config = if let Some(profile) = match &args.profile {
            // Add profile credential provider
            config.credentials_provider(
                aws_config::profile::ProfileFileCredentialsProvider::builder()
                    .profile_name(profile)
                    .build(),
            )
    } else {
        // Keep config unchanged
        config
    };
```